lewis_emulators/knr1050/interfaces/stream_interface.py

```diff
@@ -0,0 +1,122 @@
+from lewis.adapters.stream import StreamInterface
+
+from lewis.core.logging import has_log
+from lewis_emulators.utils.replies import conditional_reply
+from lewis_emulators.utils.command_builder import CmdBuilder
+
+if_connected = conditional_reply('connected')
+
+
+@has_log
+class Knr1050StreamInterface(StreamInterface):
+
+    in_terminator = '\r'
+    out_terminator = '\r'
+
+    def __init__(self):
+
+        super(Knr1050StreamInterface, self).__init__()
+        # Commands that we expect via serial during normal operation
+        self.commands = {
+            CmdBuilder(self.get_status).escape("STATUS?").build(),
+            CmdBuilder(self.start_pump).escape("RAMP:0,").int().escape(",").int().escape(",").int().escape(",").int().escape(",").int().build(),
+            CmdBuilder(self.stop_pump).escape("STOP:1,0").build(),
+            CmdBuilder(self.stop_klv).escape("STOP:2").build(),
+            CmdBuilder(self.get_pressure_limits).escape("PLIM?").build(),
+            CmdBuilder(self.set_pressure_limits).escape("PLIM:").int().escape(",").int().build(),
+            CmdBuilder(self.get_remote_mode).escape("REMOTE?").build(),
+            CmdBuilder(self.set_remote_mode).escape("REMOTE").eos().build(),
+            CmdBuilder(self.set_local_mode).escape("LOCAL").build()
+        }
+
+    def handle_error(self, request, error):
+        """
+        If command is not recognised print and error
+        Args:
+            request: requested string
+            error: problem
+        """
+        self.log.error("An error occurred at request " + repr(request) + ": " + repr(error))
+
+    @if_connected
+    def start_pump(self, flow_rate, a, b, c, d):
+        """
+        Executes ramp starting from current execution time.
+        Args:
+            flow_rate (float): the flow rate in ul/min
+            a (int): concentration A
+            b (int): concentration B
+            c (int): concentration C
+            d (int): concentration D
+        """
+        self.device.flow_rate = flow_rate
+        self.device.current_flow_rate = flow_rate
+        # crude pressure simulation
```

Comment:
You could factor this out as a function `pressure_simulation` and not need the comment.